harvard-lts/mps-embed
# Using axios instead of https so async works properly.

**Using axios instead of https so async works properly.**
* * *

**JIRA Ticket**: [LTSVIEWER-93](https://jira.huit.harvard.edu/browse/LTSVIEWER-93)

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

1. [Axios](https://axios-http.com/)
2. [Await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)

# What does this Pull Request do?
This uses the Axios node package instead of https, so that we can make the calls to LibraryCloud asynchronously. We can then use `await` commands to wait for the API call to complete before rendering the oEmbed JSON response. This fixes the issue with it displaying an old (seemingly cached) JSON response.

# How should this be tested?

A description of what steps someone could take to:
* Rebuild the container `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
* Go to https://localhost:23018/
* Click on any of the items, it should display the correct one right away without having to refresh the page in the browser.
* You can also put in other items from HDC and it should display the correct information right away without having to refresh the page in the browser. For example: https://localhost:23018/api?recordIdentifier=W20027227_URN-3:RAD.SCHL:264469


# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@enriquediaz
